tools/cache.py

```diff
@@ -82,30 +83,30 @@ def ensure(self):
 
   def erase(self):
     with self.lock():
-      if os.path.exists(self.dirname):
+      if self.dirname.exists():
         for f in os.listdir(self.dirname):
-          tempfiles.try_delete(os.path.join(self.dirname, f))
+          tempfiles.try_delete(Path(self.dirname, f))
 
   def get_path(self, name):
-    return os.path.join(self.dirname, name)
+    return Path(self.dirname, name)
 
   def get_sysroot(self, absolute):
     if absolute:
-      return os.path.join(self.dirname, 'sysroot')
+      return str(Path(self.dirname, 'sysroot'))
```

Comment:
I'm not sure of the benefit on this particular line?

Reply:
I was going for a systematic removal of `os.path.join` here.. but sure I can revert this line.